File: rgctl-tests/ecommerce-csharp/src/Ecommerce/Coolstore/OrderEndpoint.cs
using Ecommerce.Coolstore.Models;
using Ecommerce.Coolstore.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Ecommerce.Coolstore;

[ApiController]
[AllowAnonymous]
[Route("services/orders")]
public class OrderEndpoint : ControllerBase
{
    private readonly CoolstoreOrderService _orderService;

    public OrderEndpoint(CoolstoreOrderService orderService)
    {
        _orderService = orderService;
    }

    [HttpGet]
    public ActionResult<List<CoolstoreOrder>> ListAll() =>
        Ok(_orderService.GetOrders());

    [HttpGet("{orderId:long}")]
    public ActionResult<CoolstoreOrder> GetOrder(long orderId)
    {
        var order = _orderService.GetOrderById(orderId);
        if (order == null)
        {
            return NotFound();
        }

        return Ok(order);
    }
}

File: rgctl-tests/ecommerce-java/src/main/java/com/example/ecommerce/coolstore/rest/OrderEndpoint.java
package com.example.ecommerce.coolstore.rest;

import com.example.ecommerce.coolstore.model.CoolstoreOrder;
import com.example.ecommerce.coolstore.service.CoolstoreOrderService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.server.ResponseStatusException;

import java.util.List;

@RestController
@RequestMapping("/services/orders")
public class OrderEndpoint {

    private final CoolstoreOrderService orderService;

    public OrderEndpoint(CoolstoreOrderService orderService) {
        this.orderService = orderService;
    }

    @GetMapping
    public List<CoolstoreOrder> listAll() {
        return orderService.getOrders();
    }

    @GetMapping("/{orderId}")
    public CoolstoreOrder getOrder(@PathVariable long orderId) {
        return orderService
                .getOrderById(orderId)
                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
    }
}
